rio_tiler/models.py

```diff
@@ -474,6 +481,12 @@ def create_from_list(cls, data: Sequence["ImageData"]) -> "ImageData":
         )
         dataset_statistics = stats if len(stats) == len(band_names) else None
 
+        metadata = dict(
+            itertools.chain.from_iterable(
+                [img.metadata.items() for img in data if img.metadata]
+            )
+        )
```

Comment:
we merge the metadata all together.
It's assumed that for MultiBands/MultiAssets the metadata will be in form of `{"asset1": {...}}`